packages/@react-aria/grid/src/useGrid.ts

```diff
@@ -114,7 +114,7 @@ export function useGrid<T>(props: GridProps, state: GridState<T, GridCollection<
       id,
       'aria-multiselectable': state.selectionManager.selectionMode === 'multiple' ? 'true' : undefined
     },
-    state.isKeyboardNavigationDisabled ? {} : collectionProps,
+    state.collection.size === 0 || state.isKeyboardNavigationDisabled ? {} : collectionProps,
```

Comment:
Opted for putting this logic here so this behavior would apply to all grid components, not just TableView. 

Note that this would prevent the user from tabbing to the column headers to resize them if the TableView is empty, but I figured this was a non issue since the Table doesn't have row data to display. Perhaps we should remove the resize menu entirely for this case?